ci/lint/06_script.sh

```diff
@@ -11,7 +11,7 @@ set -ex
 if [ -n "$LOCAL_BRANCH" ]; then
   # To faithfully recreate CI linting locally, specify all commits on the current
   # branch.
-  COMMIT_RANGE="$(git merge-base HEAD master)..HEAD"
+  COMMIT_RANGE="$(git merge-base HEAD origin/master)..HEAD"
```

Comment:
Preserving @maflcko's comment on the previous PR:

> unrelated: Generally I'd find it better if we get rid of the commit range completely. For example, when running the whitespace linter, it seems easier to just lint all code at `HEAD`, instead of trying to guess which code was modified and only lint that.

Reply:
I remember this fixed an error complaining that there's no `master` branch. But I can retest.

Reply:
IIUC only the cirrus linter CI uses calls this script. So it's possible this commit was only useful in #29259, where I added that linter to a Github Action. I'll drop it if CI passes on https://github.com/Sjors/bitcoin/pull/30

Reply:
I dropped 0ac57f8becd10277549d1612a11b3b7c48b1fd6a.

Someone can drop `COMMIT_RANGE` altogether in a separate PR. Maybe move the incantation to a README, because it's certainly useful to know how to run the linter on each commit.